realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -150,25 +150,82 @@ public final boolean isValid() {
     }
 
     /**
-     * Determines if the current RealmObject is obtained synchronously or asynchronously (from a worker thread).
+     * Checks if the query used to find this RealmObject has completed.
+     *
+     * Objects created from async methods like {@link RealmQuery#findFirstAsync()} return a
+     * dummy object that represent the future result of the {@link RealmQuery}. It can be considered
+     * similar to a {@link java.util.concurrent.Future} in this regard.
+     *
+     * Once this method returns {@code true}, this object represents the query result even if the query
+     * didn't find any object matching the query parameters. In this case the RealmObject will
+     * become a "null" object.
+     *
+     * "Null" objects will throw if any accessor is called, so it is important to also check {@link #isValid()}
+     * before calling any methods. A common pattern is this:
+     *
+     * <pre>
+     * {@code
+     * Person person = realm.where(Person.class).findFirstAsync();
+     * person.addChangeListener(new RealmChangeListener() {
+     *      @Override
+     *      public void onChange(Person person) {
+     *          if (person.isLoaded() && person.isValid()) {
```

Comment:
Shouldn't matter though. Even if you re-use the listener, it will only be called with a loaded object. `isLoaded()` is only really useful outside the listener.